specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2021-11-01/schemaregistry.json

```
@@ -0,0 +1,441 @@
+                "description": "Serialization type for the schema being stored.",
+                "enum": [
+                  "avro"
+                ],
+                "x-ms-enum": {
+                  "name": "SerializationType",
+                  "modelAsString": true,
+                  "values": [
+                    {
+                      "value": "avro",
+                      "name": "AVRO",
+                      "description": "Avro Serialization schema type"
+                    }
+                  ]
+                }
+              },
+              "Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "Schema-Id-Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema ID."
+              },
+              "Schema-Group-Name": {
+                "type": "string",
+                "description": "References schema group."
+              },
+              "Schema-Name": {
+                "type": "string",
+                "description": "References schema name."
+              },
+              "Schema-Version": {
+                "type": "integer",
+                "description": "Version of the returned schema."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/SchemaObject"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Schema Registry service.",
+            "schema": {
+              "$ref": "#/definitions/ServiceErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/$schemagroups/{group-name}/schemas/{schema-name}": {
+      "post": {
+        "summary": "Register new schema",
+        "description": "Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.\n",
+        "x-ms-examples": {
+          "Register schema": {
+            "$ref": "./examples/OperationSchema_Register.json"
+          }
+        },
+        "operationId": "Schema_Register",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "text/plain;charset=UTF-8"
+        ],
+        "parameters": [
+          {
+            "name": "group-name",
+            "in": "path",
+            "description": "Schema group under which schema should be registered.  Group's serialization type should match the serialization type specified in the request.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "schema-name",
+            "in": "path",
+            "description": "Name of schema being registered.",
+            "required": true,
+            "type": "string"
```

You should specify a `maxLength` and `pattern` on path parameters to restrict the length and allowed set of characters for user-specified names, e.g.
```suggestion
            "type": "string",
            "maxLength": 50,
            "pattern": "^[A-Za-z0-9][A-Za-z0-9_-]*$",
```